[docs] Update Localization (#6675)
expo/expo: docs/pages/versions/unversioned/sdk/localization.md

## Usage

Let's make our app support English and Japanese.

- Install the i18n package `i18n-js`

  ```sh
  yarn add i18n-js
  ```

- Configure the languages for your app.

  ```tsx
  import * as Localization from 'expo-localization';
  import i18n from 'i18n-js';
  // Set the key-value pairs for the different languages you want to support.
  i18n.translations = {
    en: { welcome: 'Hello' },
    ja: { welcome: 'こんにちは' },
  };
  // Set the locale once at the beginning of your app.
  i18n.locale = Localization.locale;
  ```

### API Design Tips

> - You may want to only have localized text for some things but not for others things like names. In this case you can define them _once_ in your default language and reuse them with `i18n.fallbacks = true;`.

Reviewer: ```suggestion
- You may want to refrain from localizing text for certain things, like names. In this case you can define them _once_ in your default language and reuse them with `i18n.fallbacks = true;`.
```

## Constants

### Behavior

<!-- - **iOS:** Completely synchronous, changing a system language will reset the app.
- **Android:** On some versions of the Android SDK the app may not reset. In this case you can use `i18n` with `React.useEffect` and `Localization.getLocalizationAsync()` to update the React state.
  - You can do this by checking the `AppState` to tell when the app comes back into the foreground.
  - If the value of `Localization.getLocalizationAsync()` doesn't match upon re-entry then you can quickly reset the project with `Updates` API from `expo-updates`. -->

> This API is mostly synchronous and driven by constants. On iOS the constants will always be correct, on Android you should check if the locale has updated using `AppState` and `Localization.getLocalizationAsync()`. Initially the constants will be correct on both platforms, but on Android a user can change the language and return, more on this later.

Reviewer: thing you may have forgotten to uncomment the ^ changes, and delete the old blurb? 🤷‍♂
